src/Symfony/Component/Messenger/Middleware/HandleMessageInNewTransactionMiddleware.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Middleware;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\EnvelopeAwareInterface;
+use Symfony\Component\Messenger\Exception\MessageHandlingException;
+use Symfony\Component\Messenger\Middleware\Configuration\Transaction;
+
+/**
+ * Allow to configure messages to be handled in a new Doctrine transaction if using
```

Comment:
Sorry, I didn't reviewed this yet, but is the doctrine transaction the only use-case here? I think the answer is no and this has a more generic purpose, doctrine transactions being only one, right?